### DISCUSSION TITLE
GlobalScaling Parameter in loadURDF to units

### DISCUSSION THREAD

**Discussion by shivanimall** [bulletphysics/bullet3#3809] [General]
1/ I understand that we can use `globalScaling` parameter to scale objects up and down, however, is there a way to apply a specific scaling such as the longest diameter is say 1 metre? so, use this scaling factor to know how many "meters" I am scaling the object by? 

2/ Is there a way to know the size of the loaded object? or should I rely on obj file to understand this?

3/ When I change the scale of the object, how much does the base mass of object change? And hence the collision forces generated change?

Please let me know if my question is unclear. Thank you!

**Comment by erwincoumans** [comment]
Scaling is relative to the original size, so no.

You can estimate the size of an object using its bounding box (AABB). You can get the AABB through the PyBullet API.

Mass is not affected by rescaling, you need to change that manually. Inertia is recomputed based on mass and collision geometry/shape (which is affected by scaling).
